core/store/prometheus.go

```diff
@@ -0,0 +1,38 @@
+package store
+
+import (
+	"chainlink/core/assets"
+	"chainlink/core/eth"
+	"chainlink/core/logger"
+	"errors"
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var promETHBalance = promauto.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "eth_balance",
+		Help: "Each Ethereum account's balance",
+	},
+	[]string{"account"},
+)
+
+func updatePrometheusEthBalance(balance *assets.Eth, from common.Address) {
+	balanceFloat, err := approximateFloat64(balance)
+	logger.ErrorIf(err)
+	promETHBalance.WithLabelValues(from.Hex()).Set(balanceFloat)
```

Comment:
Won't this set the balance even if there is an error?

Reply:
Was this intended?  As you do not bail out early after the balance error below
```
ethBalance, linkBalance, balanceErr := txm.GetETHAndLINKBalances(tx.From)
```